### ISSUE TITLE
Limit the memory resources while running fmin2

### ISSUE THREAD

**Issue by juantorres91** [CMA-ES/pycma#238]
Hi,

Currently I'm benchmarking  DFO software including cma via fmin2. I'm running the test in  new server, at the moment we do not have a scheduler. Everything is working fine but the execution is using lots of resources, the cpu usage is 2600 % at its peak, and it seems to be using all the processors. I have tried to use  resource.setrlimit or ulimit with no effect. How can I limit cma's memory use?

Thanks,
Juan

**Comment by nikohansen** [comment]
I don't know whether the question refers to CPU or memory, as you mention both.

What comes in mind: setting
```python
import cma
cma.evolution_strategy._CMASolutionDict = cma.evolution_strategy._CMASolutionDict_empty
```
may help to reduce (both) resources for long runs and should work as long as the code is invoked without the `bounds` option (that is specifically, without genotype-phenotype transformation).

**Comment by Wukong-SCUT** [comment]
Indeed, we have also encountered such problems in our work. The main reason is that the power of the CPU will be full immediately, and the time consumption is huge (compared to other libraries). After trying the method you mentioned, the power of the CPU and has not declined. Is it caused by bounds boundary conversion?

**Comment by Wukong-SCUT** [comment]
In addition, I would like to ask you a question. In my attempt, I enlarged these values ​​with 'verb_log_expensive':False,'verb_disp':1e6,'verb_log':1e6 so that it will no longer be recorded, which improves efficiency. Also improved. I would like to ask if there is a way to turn off all records to improve efficiency? In fact, I only need the optimizer (but the current efficiency is still not high, and the CPU usage is too high). I hope you can help me solve this problem despite your busy schedule, thank you!
<img width="356" alt="屏幕截图 2024-03-23 090701" src="https://github.com/CMA-ES/pycma/assets/100810925/7f8dab94-50dc-4eca-9461-97ee6495e7b3">

**Comment by Wukong-SCUT** [comment]
> Indeed, we have also encountered such problems in our work. The main reason is that the power of the CPU will be full immediately, and the time consumption is huge (compared to other libraries). After trying the method you mentioned, the power of the CPU and has not declined. Is it caused by bounds boundary conversion?

I have solved this problem very well. It is judged that it is due to OpenMP. You only need to enter: export OMP NUM THREADS=1 in the command line before use to avoid the problem of CPU being full.

**Comment by nikohansen** [comment]
This turns out to be a known issue, see also https://github.com/numbbo/coco/issues/1919 and [the fix](https://github.com/numbbo/coco/issues/1919#issuecomment-624051032) which seems to work as well.

**Comment by Wukong-SCUT** [comment]
已收到，感谢您的来信。       ——悟空同学
